packages/peregrine/lib/Apollo/policies/index.js

```diff
@@ -143,6 +143,9 @@ const typePolicies = {
             }
         }
     },
+    CustomerPaymentTokens: {
+        keyFields: () => 'CustomerPaymentTokens'
```

Comment:
This is a clever solution, just give the collection an identity. I would normally suggest that the `PaymentToken` type needs an identity, but that may not be necessary since there's no way to mutate it. I also don't like adding these hidden requirements to queries, not sure how Apollo reacts when the field for the key identity is missing.

@sirugh - is there a technical reason we should do something like:
```javascript
PaymentToken: {
    keyFields: ({ public_hash }) => `CustomerPaymentToken:${public_hash}`
},
```

Reply:
The solution you are proposing would work on individual tokens, the one I have here is for the whole collection. I feel like adding `keyFields` to each token would be overkill unless it provides some use. I am not equipped enough to make the call about the apollo types we are using here, I am fine with what the team thinks is the right approach.

Reply:
> If one of your types defines its primary key with a field besides id or _id, you can customize how the InMemoryCache generates unique identifiers for that type. To do so, you define TypePolicy for the type. You specify all of your cache's typePolicies in the options object you provide to the InMemoryCache constructor.

This is from the [docs](https://www.apollographql.com/docs/react/caching/cache-configuration/#data-normalization). I wouldn't try to get much more creative that our use case. We only use this because we need a handle on the cache key in order to delete it. If we find that other properties are still cached and not garbage collected after the deletion of these other objects then we may need to identify PaymentTokens with keyFields and delete them too.